### PR TITLE
Refactored deploy task

### PR DESCRIPTION
**Pull request by avasquez614**
* Moved resources to a more logical dir hierarchy
* Deploy task now overwrites config files when -PoverwriteConfig=true and keeps them if false
* Upgrade task no longer deletes bin directories

Ticket craftercms/craftercms#2896 and craftercms/craftercms#2887

### CODE REVIEW

File: .gradle.log [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[INFO] Scanning for projects...
```
Comment (sumerjabri) [inline .gradle.log]: Please add this file to .gitignore and remove from PR

File: build.gradle [hunk 2]
```diff
@@ -54,6 +54,8 @@ ext {
     /** Environment Building properties **/
     authEnvDir = project.hasProperty("authoring.root") ? project.property("authoring.root") : "./crafter-authoring/"
     deliveryEnvDir = project.hasProperty("delivery.root") ? project.property("delivery.root") : "./crafter-delivery/"
+    authProcessedResourcesDir = "./tmp/crafter-authoring"
```
Comment (sumerjabri) [inline build.gradle]: Is this relative path the right way to go? Should we not use a CRAFTER_HOME env /tmp as a the folder path?

Comment (avasquez614) [reply build.gradle]: `.` is always going to be the `craftercms` project folder, Gradle makes sure to set it up when running the build. `CRAFTER_HOME` doesn't exist until it's set in the sh scripts.

File: .gradle.log (hunk 1 above)
Comment (avasquez614) [reply .gradle.log]: Will do

**Pull request merged**